libs/natparse/src/test/java/org/amshove/natparse/parsing/StatementListParserShould.java

```diff
@@ -719,14 +719,40 @@ void parseSetKeyNamed()
 	@ParameterizedTest
 	@ValueSource(strings =
 	{
-		"SET KEY ALL", "SET KEY PF2", "SET KEY PF2=PGM", "SET KEY OFF", "SET KEY ON", "SET KEY PF2=OFF", "SET KEY PF2=ON", "SET KEY PF4='SAVE'", "SET KEY PF4=#XYX", "SET KEY PF6='LIST MAP *'", "SET KEY PF2='%%'", "SET KEY PF9=' '", "SET KEY PF12=DATA 'YES'", "SET KEY PF4=COMMAND OFF", "SET KEY PF4=COMMAND ON", "SET KEY COMMAND OFF", "SET KEY COMMAND ON", "SET KEY PF1=HELP",
+		"SET KEY ALL", "SET KEY PF2", "SET KEY PF2=PGM", "SET KEY OFF", "SET KEY ON", "SET KEY PF2=OFF", "SET KEY PF2=ON", "SET KEY PA2=ON", "SET KEY PF4='SAVE'", "SET KEY PF4=#XYX", "SET KEY PF6='LIST MAP *'", "SET KEY PF2='%%'", "SET KEY PF9=' '", "SET KEY PF12=DATA 'YES'", "SET KEY PF4=COMMAND OFF", "SET KEY PF4=COMMAND ON", "SET KEY COMMAND OFF", "SET KEY COMMAND ON", "SET KEY PF1=HELP",
 		"SET KEY PF10=DISABLED", "SET KEY ENTR NAMED 'EXEC'", "SET KEY PF3 NAMED 'EXIT'", "SET KEY PF3 NAMED OFF", "SET KEY NAMED OFF", "SET KEY PF4='AP1' NAMED 'APPL1'"
 	})
 	void parseSetKeyExamples(String statement)
 	{
 		assertParsesSingleStatement(statement, ISetKeyNode.class);
 	}
 
+	@ParameterizedTest
+	@ValueSource(strings =
+	{
+		"SET KEY CLR", "SET KEY CLR=OFF", "SET KEY CLR=COMMAND ON", "SET KEY CLR=DISABLED", "SET KEY CLR=ON NAMED 'Clear'", "SET KEY CLR=PGM NAMED OFF",
+		"SET KEY DYNAMIC #XYZ", "SET KEY DYNAMIC #XYZ = HELP", "SET KEY DYNAMIC #XYZ = DATA 'Hello' NAMED 'Dynam'"
+	})
+	void parseSetKeyClrAndDynamicExamples(String statement)
+	{
+		assertParsesSingleStatement(statement, ISetKeyNode.class);
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings =
+	{
+		"PF100", "PA4", "CLRS", "PF011", "PA01", "PF25"
+	})
+	void showADiagnosticForInvalidTokensInSetKey(String length)
+	{
+		assertDiagnostic(
+			"""
+				SET KEY %s
+				""".formatted(length),
+			ParserError.UNEXPECTED_TOKEN
+		);
+	}
+
```

Comment:
You can add this testcase to have a regression test about the comment I've previously written:

```java
	@Test
	void notMistakeAnAssignmentAsSetKeyOperand()
	{
		var statementList = assertParsesWithoutDiagnostics("""
			SET KEY PF5 NAMED 'PHON'
			#VAR := 'Hi'
			""");

		assertThat(statementList).hasSize(2);
		assertNodeType(statementList.statements().get(0), ISetKeyNode.class);
		assertNodeType(statementList.statements().get(1), IAssignmentStatementNode.class);
	}

```